### ISSUE TITLE
Regression in v3.3.1 with parsing nav element

### ISSUE THREAD

**Issue by majora2007** [vers-one/EpubReader#94]
I'm getting the following error upon opening the book, with options.
```
VersOne.Epub.Epub3NavException: EPUB parsing error: 'nav' element in the navigation file does not contain a required 'ol' element.
   at VersOne.Epub.Utils.TaskExtensionMethods.ExecuteAndUnwrapAggregateException[T](Task`1 task)
   at VersOne.Epub.Internal.BookRefReader.OpenBook(String filePath)
   at VersOne.Epub.EpubReader.OpenBook(String filePath, EpubReaderOptions epubReaderOptions)
```

```
PackageReaderOptions = new PackageReaderOptions
        {
            IgnoreMissingToc = true
        }
```

This was working prior to v3.3.0. Now it is throwing an exception. 
Here is code from the nav.xhtml 
```
<nav epub:type="toc" id="toc">
    <h1>Table of Contents</h1>
    <ol>
      <li>
        <a href="../Text/copyright.xhtml">Copyright</a>
      </li>
      <li>
        <a href="../Text/toc.xhtml">Contents</a>
      </li>
```

**Comment by vers-one** [comment]
Hmm, this `<nav>` element should be fine. Does the *nav.xhtml* file have other `<nav>`s by any chance?

**Comment by majora2007** [comment]
Great find, yes there are!

**Comment by vers-one** [comment]
I added a few more parsing validations in EpubReader 3.3.0 to make [Nullable reference type annotations](https://github.com/vers-one/EpubReader/pull/71) possible. In case of `Epub3Nav.Ol` there were two options: I could make it nullable (i.e. `Epub3NavOl?` type) but that would be against the EPUB specification which says that [there should be](https://www.w3.org/publishing/epub32/epub-packages.html#sec-package-nav-def-model) exactly one `<ol>` inside a `<nav>` and the applications would always have to check if its value is not `null` when they access this property. Or I could make it non-nullable (which aligns with the specification and which is what I chose to do) but the compiler immediately spotted that the parsed value can be `null` and raised an error. So the reason I added this parsing validation is to make sure that the `Epub3Nav.Ol` property will never get a `null` value.

I plan to add [configuration options](https://github.com/vers-one/EpubReader/issues/95) to control all validation checks but it will probably take a few weeks to implement all of them.

**Comment by majora2007** [comment]
I for one, really enjoy the nullable support as I had ran into a few instances where I didn't expect something to be null and it was. I've subscribed to your configuration options issue. I will just update our wiki with some of the issues your library throws and how my users can fix it. 

The amount of bad epubs is shocking. In Kavita, the software that uses this library, I have to balance the number of hacks to implement versus errors to throw. I've never seen a format so badly managed, even by the publishers themselves. Keys are never correct, missing toc, etc. It's just a mess.
